cics-bundle-maven-plugin/src/main/java/com/ibm/cics/bundlegen/DefineGraph.java

```diff
@@ -0,0 +1,145 @@
+package com.ibm.cics.bundlegen;
+
+/*-
+ * #%L
+ * CICS Bundle Maven Plugin
+ * %%
+ * Copyright (C) 2019 IBM Corp.
+ * %%
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * #L%
+ */
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DefineGraph {
```

Comment:
Considered using a library for this rather than implementing our own DAG?  Something like https://jgrapht.org/javadoc/org/jgrapht/graph/DirectedAcyclicGraph.html maybe?

Reply:
Not looked at the licence, or picked that one for any particular reason!

Reply:
Seems like it'd be possible to use Guava's `common.graph` which doesn't directly support acyclic graphs, but does have a `hasCycle` method we can use to detect whether the graph is acyclic: https://github.com/google/guava/issues/2609

Reply:
Yeah, looked at those two but it didn't seem worth the 1MB/2.5MB install weight plus OSS process when all we need is a very simple graph.

Reply:
Ok, fine by me!  Figured I would check!